.github/workflows/comment-trigger.yml

```diff
@@ -59,13 +59,19 @@ jobs:
           || contains(github.event.comment.body, '/ci-run-k8s')
       )
     steps:
+      - name: Generate authentication token
+        id: generate_token
+        uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92
+        with:
+          app_id: ${{ secrets.GH_APP_DATADOG_VECTOR_CI_APP_ID }}
+          private_key: ${{ secrets.GH_APP_DATADOG_VECTOR_CI_APP_PRIVATE_KEY }}
       - name: Get PR comment author
         id: comment
         uses: tspascoal/get-user-teams-membership@v2
         with:
           username: ${{ github.actor }}
           team: 'Vector'
-          GITHUB_TOKEN: ${{ secrets.GH_PAT_ORG }}
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
```

Comment:
Nice!

A potential future enhancement- we could drop this into a re-usable workflow, and have the token be an output of it, since we do it in a few places.
Could even have an optional parameter to that workflow to do the team membership check, which would populate an output as well, so we wouldn't have to repeat these two job steps throughout.